docs/reference/settings/security-settings.asciidoc

```diff
@@ -1947,6 +1947,311 @@ include::{es-repo-dir}/settings/common-defs.asciidoc[tag=ssl-supported-protocols
+provided by the JWT claims. Defaults to `true`.
+// end::jwt-populate-user-metadata-tag[]
+
+// tag::jwt-client-authentication-type-tag[]
+`client_authentication.type` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Specifies whether to use `sharedsecret`, `ssl`, or `none` authentication
+to incoming client requests. Defaults to `sharedsecret`.
+// end::jwt-client-authentication-type-tag[]
+
+`authorization_realms`::
+(<<static-cluster-setting,Static>>)
+The names of the realms that should be consulted for delegated authorization.
+If this setting is used, then the JWT realm does not perform role
+mapping and instead loads the user from the listed realms.
+See <<authorization_realms>>.
+
+`cache.ttl`::
+(<<static-cluster-setting,Static>>)
+Specifies the time-to-live for cached user entries. A user and a hash of its
+credentials are cached for this period of time. Use the
+standard {es} <<time-units,time units>>).
+Defaults to `20m`.
+
+`cache.max_users`::
+(<<static-cluster-setting,Static>>)
+Specifies the maximum number of user entries that the cache can contain.
+Defaults to `100000`.
+
+// tag::jwt-http-connect-timeout-tag[]
+`http.connect_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the timeout until a connection
+ is established. A value of zero means the timeout is not used. Defaults to `5s`.
+// end::jwt-http-connect-timeout-tag[]
+
+// tag::jwt-http-read-timeout-tag[]
+`http.connection_read_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the timeout used when
+requesting a connection from the connection manager. Defaults to `5s`
+// end::jwt-http-read-timeout-tag[]
+
+// tag::jwt-http-socket-timeout[]
+`http.socket_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the socket timeout (SO_TIMEOUT)
+in milliseconds, which is the timeout for waiting for data or, put differently,
+a maximum period inactivity between two consecutive data packets). Defaults to
+`5s`.
+// end::jwt-http-socket-timeout[]
+
+// tag::jwt-http-max-connections-tag[]
+`http.max_connections` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the maximum number of
+connections allowed across all endpoints.
+// end::jwt-http-max-connections-tag[]
+
+// tag::jwt-http-max-endpoint-connections-tag[]
+`http.max_endpoint_connections` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the maximum number of
```

Comment:
Controls the behavior of the http client used for fetching the JSON Web Key Set
from remote a URL. Specifies the maximum number of
```